### PR TITLE
paychecks are now a % of the budget instead of fixed numbers

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
tweak: paychecks are a % of budget instead of fixed numbers
/:cl:

money being 99% in budget i think was always dumb and if we remove budget  its even more dumb, now people actually get money depending on how many are in the department and how much money they "make" 

inb4 % of money arent always the same when they get payday, this only makes ppl who latejoin get paid less rest is rng fuck em

### CODE REVIEW

File: code/modules/economy/account.dm [hunk 1]
```diff
@@ -109,4 +109,4 @@
 /datum/bank_account/remote // Bank account not belonging to the local station
 	add_to_accounts = FALSE
 
-#undef DUMPTIME
+#undef DUMPTIME
```
Comment (Urumasi) [inline code/modules/economy/account.dm]: How did travis not scream here, missing EOL

File: code/modules/economy/account.dm [hunk 2]
```diff
@@ -54,7 +54,7 @@
 	else
 		var/datum/bank_account/D = SSeconomy.get_dep_account(account_job.paycheck_department)
 		if(D)
-			if(!transfer_money(D, money_to_transfer))
+			if(!transfer_money(D, round(money_to_transfer*D.account_balance*0.01,1)))
```
Comment (optimumtact) [inline code/modules/economy/account.dm]: `if(!transfer_money(D, round(money_to_transfer*(D.account_balance*0.01),1)))`

**Comment by Cyprex** [comment]
I've actually been thinking about this for a bit, how do you feel a system using weights instead of just paying a fixed percentage of money?

**Comment by Tlaltecuhtli** [comment]
paying in kilograms? i m not american

**Comment by Cyprex** [comment]
Let's say we distribute 10% of the budget, 1000 credits. Now, HoS might have 10 weight for payout, officer 5 and detective 6.
We have 1 HoS and 4 Officers, for a total weight of 4*5+10 = 30. HoS is 10 weight out of 30, or a third, so he gets a third of the 1000 credits. Officers get (5 / 30) * 1000 = ~166. Instead of paying out a fixed percentage of the defense budget you could also make it depend on weight, for example by saying there's 90 weight that won't get paid out. I could likely code something like this and PR it to your branch if you want.

**Comment by Tlaltecuhtli** [comment]
whats the difference

**Comment by optimumtact** [comment]
is this ready, did you test it?

**Comment by Tlaltecuhtli** [comment]
it does whats supposed to do

**Comment by willox** [comment]
will this mean that people in a department all earn less money depending on who is paid first during a single payday (because the budget will be lowered each time somebody is paid)?

**Comment by optimumtact** [comment]
Can someone take this over? it's crucial now that dept cards are gone

**Comment by Tlaltecuhtli** [comment]
are any changes needed or what?

**Comment by optimumtact** [comment]
it to actually work for a start

File: code/modules/economy/account.dm [hunk 3]
```diff
@@ -50,11 +50,11 @@
 /datum/bank_account/proc/payday(amt_of_paychecks, free = FALSE)
 	var/money_to_transfer = account_job.paycheck * amt_of_paychecks
 	if(free)
-		adjust_money(money_to_transfer)
+		adjust_money(money_to_transfer*10)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/economy/account.dm]: ???

Comment (Tlaltecuhtli) [reply code/modules/economy/account.dm]: PAYCHECK_HARD (read as 4) * amt_of_paychecks (read as 5) * 10

Comment (ExcessiveUseOfCobblestone) [reply code/modules/economy/account.dm]: looking at hard:
old: `moneytransfer`(40*5) = 200
new: `moneytransfer`(4*10)*10=400

am I missing something?

**Pull request merged**